typespec-tests/tsp/patch.tsp

```diff
@@ -23,16 +26,66 @@ model Resource {
   name: string;
 
   description?: string;
-  type: string;
+  map: Record<InnerModel>;
+  longValue?: int64;
+  intValue?: int32;
+  enumValue?: "a" | "b" | "c";
+
+  @projectedName("json", "wireNameForInnerModelProperty")
+  innerModelProperty?: InnerModel;
+}
+
+model InnerModel {
+  name: string;
+}
+
+@doc("This is base model for polymorphic multiple levels inheritance with a discriminator.")
+@discriminator("kind")
+@resource("resources")
+model Fish {
+  @visibility("read")
+  id: string;
+
+  @key
+  @visibility("read")
+  name: string;
+
+  age: int32;
+  color?: string;
+}
+
+@doc("The second level model in polymorphic multiple levels inheritance and it defines a new discriminator.")
+@discriminator("sharktype")
+model Shark extends Fish {
+  kind: "shark";
+  sharktype: string;
+}
+
+@doc("The second level model in polymorphic multiple levels inheritance which contains references to other polymorphic instances.")
+model Salmon extends Fish {
+  kind: "salmon";
+  friends?: Fish[];
+  hate?: Record<Fish>;
+  partner?: Fish;
+}
+
+@doc("The third level model SawShark in polymorphic multiple levels inheritance.")
+model SawShark extends Shark {
+  sharktype: "saw";
+}
+
+@doc("The third level model GoblinShark in polymorphic multiple levels inheritance.")
+model GoblinShark extends Shark {
+  sharktype: "goblin";
 }
```

Comment:
question: Do we support nested polymorphic (kind -> sharktype) in stream-style (I remember it is not supported in Jackson)?

Reply:
We need to pass in `sharktype` in constructor currently, and it will do correct serialization. May not be the expected 'support'.
I have removed the nested hierarchy case. Only keep simple inheritance.